### PR TITLE
fix: improve version check to work for backports

### PR DESCRIPTION
**Pull request by theseion**

### CODE REVIEW

**Comment by theseion** [comment]
Sorry for the noise @airween. It should be good now.

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -43,13 +43,43 @@ jobs:
         run: |
           git remote add upstream https://github.com/coreruleset/coreruleset
           git fetch --tags upstream
-
+          git fetch upstream main
+          echo ${{ github.base_ref }}
+          if [ -n "${{ github.base_ref }}" ]; then
```
Comment (airween) [inline .github/workflows/lint.yaml:48]: Is there any chance that variable `${{ github.base_ref }}` is empty? What happens if that's it?

File: .github/workflows/lint.yaml [hunk 2]
```diff
@@ -43,13 +43,43 @@ jobs:
...
+            git fetch upstream "${{ github.base_ref }}"
```
Comment (airween) [inline .github/workflows/lint.yaml:49]: Does this necessary? Later (in line 56) you check the same condition again.

File: util/crs-rules-check/rules-check.py [hunk 3]
```diff
@@ -850,7 +850,7 @@ def generate_version_string():
         # if no --version/-v was given, get version from git describe --tags output
         crsversion = generate_version_string()
```
Comment (airween) [inline util/crs-rules-check/rules-check.py:851]: The script has a `generate_version_string()` method which produces a similar version string that you make outside of the script, [here](https://github.com/coreruleset/coreruleset/blob/c57c458da6158cd747c0e4c193de9729748de67f/util/crs-rules-check/rules-check.py#L811-L827). May be we should replace the logic there by yours. Or could we leave it as is? What do you think?

**Review by airween — COMMENTED** [review]
Added few questions.

**Comment by airween** [comment]
> Sorry for the noise @airween. It should be good now.

No worries, thanks for taking this. I added a few questions, but generally it looks good to me.

File: .github/workflows/lint.yaml (hunk 1 above)
Comment (theseion) [reply .github/workflows/lint.yaml:48]: Yes, in which case `version_flag` will be empty.

File: .github/workflows/lint.yaml (hunk 2 above)
Comment (theseion) [reply .github/workflows/lint.yaml:49]: Yes, those are two different scripts, they know nothing of each other.

File: util/crs-rules-check/rules-check.py (hunk 3 above)
Comment (theseion) [reply util/crs-rules-check/rules-check.py:851]: Yes, we could do that, but then we would need to pass in the base ref (the "target branch"). And even then, part of the logic will continue to live in the pipeline (fetching of the base ref, tags, etc.). The only logic that makes sense to have in the script, IMO, is the creation of the expected version string from the tag.

Something like this:
1. if `-v` is empty, don't perform the version check
2. if `-v` is not empty, `args.version` contains the Git tag
3. if `-v` is not empty, convert Git tag to expected version and run check

I'd then also rename `-v` to `check-version`, for example.

**Review by airween — APPROVED** [review]

**Pull request merged**